### PR TITLE
Make `btsieve` traits use async/await 

### PR DESCRIPTION
**Pull request by tcharding**
We want async/await all over, make it so in `btsieve`.

This PR makes the following traits async: `ReceiptByHash`, `LatestBlock`, `BlockByHash`.

Removes the `impl_block_by_hash` macro.

Fixes: #1879

### CODE REVIEW

File: cnd/src/btsieve/bitcoin/cache.rs [hunk 1]
```diff
@@ -28,35 +24,55 @@ impl<C> Cache<C> {
...
         }
-        .boxed()
-        .compat();
 
-        Box::new(future)
+        Ok(block)
     }
 }
 
-impl_block_by_hash!();
```
Comment (luckysori) [inline cnd/src/btsieve/bitcoin/cache.rs]: Quick look on my phone, but it looks like you stopped using the macro yet you didn't delete it and actually updated it to be async. Am I wrong?

Comment (tcharding) [reply cnd/src/btsieve/bitcoin/cache.rs]: Thanks @luckysori, you are right.  Deleted it now and force pushed.  (unrelated note: Are we aiming for correct discreet patches on PRs yet?)

Comment (thomaseizinger) [reply cnd/src/btsieve/bitcoin/cache.rs]: Aim for patches that yield a Rust binary :)

File: cnd/src/btsieve/bitcoin/cache.rs [hunk 2]
```diff
@@ -28,35 +24,55 @@ impl<C> Cache<C> {
...
+#[async_trait]
+impl<C> BlockByHash for Cache<C>
+where
+    C: BlockByHash<Block = Block, BlockHash = Hash> + Clone,
+{
+    type Block = Block;
+    type BlockHash = BlockHash;
+
+    async fn block_by_hash(&mut self, block_hash: Self::BlockHash) -> anyhow::Result<Self::Block> {
+        let mut connector = self.connector.clone();
```
Comment (thomaseizinger) [inline cnd/src/btsieve/bitcoin/cache.rs]: You shouldn't need to clone the connector :)

**Pull request merged**